### PR TITLE
AMP additional sd

### PR DESCRIPTION
**Pull request by nicl**
* add WebPage as structured data
* improve NewsArticle type (add mainEntityOfPage)
* add trailText to model to be used as meta description

### CODE REVIEW

File: article/app/model/dotcomponents/DotcomponentsDataModel.scala [hunk 1]
```diff
@@ -304,13 +255,31 @@ object DotcomponentsDataModel {
...
+      val authors = article.tags.contributors.map(contributor => {
+        Person(
+          name = contributor.name,
+          sameAs = contributor.metadata.webUrl,
+        )
+      })
+
+      List(
+        NewsArticle(
+          `@id` = Configuration.amp.baseUrl + article.metadata.id,
```
Comment (gtrufitt) [inline article/app/model/dotcomponents/DotcomponentsDataModel.scala]: It's not clear (to me..) from reading the docs what `@id` in the newsArticle context is used for, however - is this purely `AMP` or is this structured data that will be used for the main article pages in DC too? Feel free to just reply with a link to a page on schema.org or whatever that explains!

File: article/app/model/dotcomponents/DotcomponentsDataModel.scala [hunk 2]
```diff
@@ -304,13 +255,31 @@ object DotcomponentsDataModel {
...
+            ImgSrc(mainImageURL, Item1200),
+          ),
+          author = authors,
+          datePublished = article.trail.webPublicationDate.toString(),
+          dateModified = article.fields.lastModified.toString(),
+          headline = article.trail.headline,
+          mainEntityOfPage = article.metadata.webUrl,
+        ),
+        WebPage(
+          `@id` = article.metadata.webUrl,
```
Comment (gtrufitt) [inline article/app/model/dotcomponents/DotcomponentsDataModel.scala]: Yeah, this is what I expected in this field but don't know how `newsArticle` differs

File: common/app/views/support/ImageProfile.scala [hunk 3]
```diff
@@ -117,6 +117,12 @@ object Video700 extends VideoProfile(width = Some(700))
 object Video1280 extends VideoProfile(width = Some(1280))
 object GoogleStructuredData extends ImageProfile(width = Some(300), height = Some(300)) // 1:1
 
+// Used for AMP image structured data - see
```
Comment (gtrufitt) [inline common/app/views/support/ImageProfile.scala]: 🎊

File: article/app/model/dotcomponents/DotcomponentsDataModel.scala [hunk 4]
```diff
@@ -342,16 +311,17 @@ object DotcomponentsDataModel {
...
       webURL = article.metadata.webUrl,
       shouldHideAds = article.content.shouldHideAdverts,
       hasStoryPackage = articlePage.related.hasStoryPackage,
       hasRelated = article.content.showInRelated,
       isCommentable = article.trail.isCommentable,
-      article.metadata.commercial,
-      article.content.starRating
+      commercialProperties = article.metadata.commercial,
+      starRating = article.content.starRating,
+      trailText = article.trail.fields.trailText.getOrElse("")
```
Comment (gtrufitt) [inline article/app/model/dotcomponents/DotcomponentsDataModel.scala]: I think you've added trailText as a `description` meta tag, what was the reason to do that vs `description` on the json+ld as described as optional field on https://developers.google.com/search/docs/data-types/article#amp

File: common/app/model/meta/LinkedData.scala [hunk 5]
```diff
@@ -29,8 +38,8 @@ object Logo {
 }
 
 case class Guardian(
-  override val `@type`: String = "Organization",
-  override val `@context`: String = "http://schema.org",
+  `@type`: String = "Organization",
+  `@context`: String = "http://schema.org",
```
Comment (gtrufitt) [inline common/app/model/meta/LinkedData.scala]: minor but `https` 🔒 (few in the PR)

File: common/app/model/meta/LinkedData.scala [hunk 6]
```diff
@@ -29,8 +38,8 @@ object Logo {
...
   `@id`: String = "https://www.theguardian.com#publisher",
   name: String = "The Guardian",
   url: String = "http://www.theguardian.com/",
```
Comment (gtrufitt) [inline common/app/model/meta/LinkedData.scala]: `https`?

File: common/app/model/meta/LinkedData.scala [hunk 7]
```diff
@@ -62,13 +78,80 @@ object PotentialAction {
...
+)
+
+object Person {
+  implicit val formats: OFormat[Person] = Json.format[Person]
+}
+
+case class IsPartOf(
+  `@type`: List[String] = List("CreativeWork", "Product"),
+  name: String = "The Guardian",
+  productID: String = "theguardian.com:basic"
```
Comment (gtrufitt) [inline common/app/model/meta/LinkedData.scala]: Maybe to review in later schema work but I don't get product makes sense here...

**Review by gtrufitt — APPROVED** [review]

File: article/app/model/dotcomponents/DotcomponentsDataModel.scala (hunk 1 above)
Comment (nicl) [reply article/app/model/dotcomponents/DotcomponentsDataModel.scala]: good point! This is AMP specific. I did this to mirror current behaviour. If we use the regular URL here then the NewsArticle and WebPage structured data fields get merged (as same ID), which I think from a SD sense is fine, but I was wary of differing from PROD here as I'd prefer to get the same behaviour first and then improve. Having said that, I haven't fully mirrored prod here.

File: common/app/model/meta/LinkedData.scala (hunk 6 above)
Comment (nicl) [reply common/app/model/meta/LinkedData.scala]: hmm interesting. Yes that should be changed!

File: common/app/model/meta/LinkedData.scala (hunk 7 above)
Comment (nicl) [reply common/app/model/meta/LinkedData.scala]: Yeah I don't really know much about this bit to be honest.

File: article/app/model/dotcomponents/DotcomponentsDataModel.scala (hunk 4 above)
Comment (nicl) [reply article/app/model/dotcomponents/DotcomponentsDataModel.scala]: So, Google say you should always add a description meta tag. It's unclear if that allows for SD, but it doesn't mention that - https://support.google.com/webmasters/answer/79812?hl=en - it seems like they require you to have a title meta tag and description for all articles. But it is definitely unclear.

See, e.g., https://support.google.com/webmasters/answer/35624?hl=en for an example of where they say `Make sure that every page on your site has a meta description.`

File: common/app/model/meta/LinkedData.scala (hunk 5 above)
Comment (nicl) [reply common/app/model/meta/LinkedData.scala]: I had a look at this and am a bit unsure if/how this matters. All of the examples I can find (including on schema.org) use the non-https version so was thinking of sticking with that.

Comment (nicl) [reply common/app/model/meta/LinkedData.scala]: Update: actually, the Google examples do use https and I agree this seems logical! Have changed.

**Pull request merged**